src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs

```diff
@@ -227,6 +449,95 @@ public void Dispose()
             }
         }
 
+        private static bool CompareTags(IEnumerable<KeyValuePair<string, object?>>? tags1, IEnumerable<KeyValuePair<string, object?>>? tags2)
```

Comment:
Same comments here as on the other one.  It's unfortunate we're duplicating the code.  Even if we need to build it into two different assemblies, can we at least have a single shared copy of the source in a static helper file that's built into both assemblies?